packages/netsuite-adapter/src/filters/instance_references.ts

```diff
@@ -110,14 +131,15 @@ const replaceReferenceValues = async (
     if (!_.isString(value)) {
       return value
     }
-    const capturedServiceId = captureServiceId(value)
-    if (_.isUndefined(capturedServiceId)) {
+    const serviceIdInfo = captureServiceIdInfo(value)
+    if (_.isUndefined(serviceIdInfo)) {
       return value
     }
-    const elemID = fetchedElementsServiceIdToElemID[capturedServiceId]
-      ?? elementsSourceServiceIdToElemID[capturedServiceId]
+    const elemID = fetchedElementsServiceIdToElemID[serviceIdInfo[CAPTURED_SERVICE_ID]]
+      ?? elementsSourceServiceIdToElemID[serviceIdInfo[CAPTURED_SERVICE_ID]]
 
-    if (_.isUndefined(elemID)) {
+    const type = serviceIdInfo[CAPTURED_TYPE]
+    if (_.isUndefined(elemID) || (type && type !== elemID.typeName)) {
```

Comment:
nit: IIUC no need to check `type`
```suggestion
    if (_.isUndefined(elemID) || type !== elemID.typeName) {
```

Reply:
We need here to check whether the type exists.
According to your suggestion, this condition will be evaluated to `true` whenever the type is `undefined` which is the majority of the cases, while we would like it to be `false` in this case.